### ISSUE TITLE
Direction enum behaves oddly

### ISSUE THREAD

**Issue by SocraticPhoenix** [SpongePowered/SpongeAPI#1265]
Code:
```java
    public static void main(String[] args) {
        System.out.println(Direction.NORTHWEST.toVector3d());
        System.out.println(Direction.NORTHEAST.toVector3d());
        System.out.println(Direction.SOUTHEAST.toVector3d());
        System.out.println(Direction.SOUTHWEST.toVector3d());
    }
```

Output:
```plain
(-0.7071067811865475, 0.0, -0.7071067811865475)
(0.7071067811865475, 0.0, -0.7071067811865475)
(0.7071067811865475, 0.0, 0.7071067811865475)
(-0.7071067811865475, 0.0, 0.7071067811865475)
```

I've narrowed down the reason my precious tree branches aren't starting in their correct locations to the above "bug." The cause of this is [this line](https://github.com/SpongePowered/SpongeAPI/blob/master/src/main/java/org/spongepowered/api/util/Direction.java#L117).

The problem I have is that Vector3d#toInt floors the values, so all the above become (0, 0, 0). I feel like this isn't intended behavior, so I'm opening an issue here.

**Comment by Deamon5550** [comment]
These direction vectors should absolutely be normalized. I'm not quite sure what you're asking us to do here?

**Comment by JBYoshi** [comment]
@SocraticPhoenix Here's something you can try: Use a `Vector3d

**Comment by SocraticPhoenix** [comment]
@Deamon5550 
Why is it necessary to normalize them?

**Comment by SocraticPhoenix** [comment]
@Deamon5550 
Okay, I understand that the enum will not change. But the problem is that Vector3d#toInt floors values, therefore, even if I use the directions to convert it around in floating point, the floating points will ultimately be truncated. Perhaps it would be possible to add a getter for the block-position vector?

**Comment by Deamon5550** [comment]
If you don't want it to floor then use `Vector3d#ceil` and you'll get a vector which is `(±1, ±1)`.

**Comment by JBYoshi** [comment]
> the floating points will ultimately be truncated.

True. However, by leaving them in the calculations, they still make a difference.

In addition, a block-position vector could end up skipping blocks. Take `NORTH_NORTHEAST` as an example: it's impossible to make an accurate `int`  out of that without skipping blocks.
